### PR TITLE
feat(auth): Add react versions of login / register forms (no routes)

### PR DESCRIPTION
**Pull request by evanpurkhiser**
This adds an initial version of the auth / register views.

These components will not yet be rendered anywhere as they have not yet been included in the route tree.

Pretty much matches the current login pages
image

Sentry w/ registration:
image

### CODE REVIEW

File: src/sentry/static/sentry/app/views/auth/loginForm.jsx [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+import PasswordField from 'app/components/forms/passwordField';
+import SentryTypes from 'app/sentryTypes';
+import TextField from 'app/components/forms/textField';
+import space from 'app/styles/space';
+
+// TODO(epurkhiser): The abstraction here would be much nicer if we just
+// exposed a configuration object telling us what auth providers there are.
+const LoginProviders = ({vstsLoginLink, githubLoginLink}) => (
+  <ProviderWrapper>
+    <ProviderHeading>External Account Login</ProviderHeading>
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/auth/loginForm.jsx]: Should this have `t()`?

**Review by markstory — COMMENTED** [review]
Are you thinking of landing this and then landing APIs / routes for these flows separately?

**Comment by EvanPurkhiser** [comment]
> Are you thinking of landing this and then landing APIs / routes for these flows separately?

Yes

File: src/sentry/static/sentry/app/views/auth/registerForm.jsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+    return (
+      <Form
+        initialData={{subscribe: true}}
+        submitLabel={t('Continue')}
+        onSubmit={this.handleSubmit}
+        footerClass={formFooterClass}
+        errorMessage={errorMessage}
+        extraButton={
+          <PrivacyPolicyLink href="https://sentry.io/privacy/" target="_blank">
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/auth/registerForm.jsx]: `target="_blank"` is default for `<ExternalLink≥`

File: src/sentry/static/sentry/app/sentryTypes.jsx [hunk 3]
```diff
@@ -42,6 +42,14 @@ export const User = PropTypes.shape({
   username: PropTypes.string,
 });
 
+export const AuthConfig = PropTypes.shape({
+  canRegister: PropTypes.bool,
+  serverHostname: PropTypes.string,
+  hasNewsletter: PropTypes.bool,
```
[the remark this answers is not in the document]
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/sentryTypes.jsx]: sentry has a whole module called `newsletter` so I guess not

File: src/sentry/static/sentry/app/views/auth/registerForm.jsx (hunk 2 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/auth/registerForm.jsx]: Fixed

**Review by billyvg — APPROVED** [review]
lgtm

**Pull request merged**